### ISSUE TITLE
Error when using cookiefile

### ISSUE THREAD

**Issue by jakeman2048** [wummel/linkchecker#539]
When attempting to use a cookiefile with linkchecker 9.3:

<pre><code>AttributeError: 'NoneType' object has no attribute 'from_file'
      cookies.from_file = <local> !AttributeError: 'NoneType' object has no attribute 'from_file'    line: for cookie in cookies.from_file(config["cookiefile"]):
      cookies.from_file = <local> !AttributeError: 'NoneType' object has no attribute 'from_file'System info:  File "/usr/local/lib/python2.7/dist-packages/linkcheck/director/aggregator.py", line 47, in new_request_session

Released on: 16.7.2014
Python 2.7.3 (default, Mar 13 2014, 11:03:55)
[GCC 4.7.2] on linux2
System info:
LinkChecker 9.3</code></pre>

Looks like it might have something to do with this (line 67 of linkcheck/director/aggregator.py): <pre><code>self.cookies = None</code></pre>

**Comment by kqr** [comment]
I also have this problem. Were there ever any solution?

**Comment by kvz** [comment]
Same problem here

**Comment by kvz** [comment]
Downgrading to 9.2 fixed it for me. It did raise another issue though:

```
 lines = <local> ['# Netscape HTTP Cookie File', '# http://curl.haxx.se/rfc/cookie_spec.html', '# This file was generated by libcurl! Edit at your own risk.'], _[0]: {len = 27}
  File "/srv/transloadit-crm/current/stack/lib/python2.7/site-packages/linkcheck/cookies.py", line 58, in from_headers
    line: raise ValueError("Required header 'Host:' missing")
```

On a curl generated cookie that starts with:

```bash
# Netscape HTTP Cookie File
# http://curl.haxx.se/rfc/cookie_spec.html
# This file was generated by libcurl! Edit at your own risk.

```

Seems LinkChecker should also ignore comments & whitespace

**Comment by kartagis** [comment]
@kvz how did you downgrade to 9.2? Also, could you send me the 9.2 version?

**Comment by kvz** [comment]
I saved `LinkChecker==9.2` in my `requirements.txt`

**Comment by kartagis** [comment]
Thanks.

**Comment by kartagis** [comment]
@kvz I saved `LinkChecker==9.2` in my `requirements.txt` too like you did, then did `python setup.py sdist --manifest-only;python setup.py build;sudo setup.py install` and linkchecker ended up being 9.4. How did you pull that off?

**Comment by candlerb** [comment]
Same problem with 9.3 (installed via "apt-get install linkchecker" on Ubuntu 18.04)

Attempted workaround:

```
apt-get remove --purge linkchecker
apt-get autoremove --purge
apt-get install python-pip
pip install linkchecker==9.2 requests==2.2.0
```

But it then fails with:

```
This program requires Python requests 2.2.0 or later.
```

This message comes from `/usr/local/lib/python2.7/dist-packages/linkcheck/__init__.py`

```
import requests
if requests.__version__ <= '2.2.0':
    raise SystemExit("This program requires Python requests 2.2.0 or later.")
```

So actually it requires 2.2.1 or later.  And it fails with 2.18.4 (default Ubuntu 18.04 version) because:

```
>>> '2.18.4' < '2.2.0'
True
```

Workaround is to install 2.2.1 :-(

But it still doesn't work; `cookies.from_file` returns a nested list of cookies, whereas `director.aggregator.new_request_session` calls `requests.cookies.merge_cookies` which requires either a dict or CookieJar instance (and fails silently for other types of object)

Here is a quick frig to fix the worst of it:

```
--- cookies.py.orig	2020-11-25 11:49:14.665475155 +0000
+++ cookies.py	2020-11-25 12:07:23.251174002 +0000
@@ -36,12 +36,12 @@
             line = line.rstrip()
             if not line:
                 if lines:
-                    entries.append(from_headers("\r\n".join(lines)))
+                    entries.extend(from_headers("\r\n".join(lines)))
                 lines = []
             else:
                 lines.append(line)
         if lines:
-            entries.append(from_headers("\r\n".join(lines)))
+            entries.extend(from_headers("\r\n".join(lines)))
         return entries


```

```
--- director/aggregator.py.orig	2020-11-25 11:52:41.669800926 +0000
+++ director/aggregator.py	2020-11-25 11:58:09.426317182 +0000
@@ -39,7 +39,7 @@
     # XXX proxies
     if config["cookiefile"]:
         for cookie in cookies.from_file(config["cookiefile"]):
-            session.cookies = requests.cookies.merge_cookies(session.cookies, cookie)
+            session.cookies = requests.cookies.merge_cookies(session.cookies, {cookie.name: cookie.value})
     return session


```

But as far as I can see, this has never worked.
